src/draw/lv_draw_buf.c

```diff
@@ -70,9 +70,15 @@ void * lv_draw_buf_align(void * data, lv_color_format_t color_format)
     else return NULL;
 }
 
-void lv_draw_buf_invalidate_cache(void * buf, uint32_t stride, lv_color_format_t color_format, const lv_area_t * area)
+void lv_draw_buf_invalidate_cache(lv_draw_buf_t * draw_buf, const lv_area_t * area)
 {
-    if(handlers.invalidate_cache_cb) handlers.invalidate_cache_cb(buf, stride, color_format, area);
+    if(handlers.invalidate_cache_cb) {
+        LV_ASSERT_NULL(draw_buf);
+        const lv_image_header_t * header = &draw_buf->header;
+        lv_area_t full = {0, 0, header->w - 1, header->h - 1};
```

Comment:
Well, I don't know how good the optimizers of C compilers are, but it would be better to "help" them by moving `lv_area_t full` into the `if` statement, so that it will be processed only when `area == NULL`, and not every time we call the function.

Btw thx for simplifying this invalidate cache function to require only a draw_buf, instead of previous buf + stride + color_format...

Reply:
The definition of `lv_area_t full` cannot move to the `if` statement because it's on stack and will be out of scope outside of the `if` statement.
I have made the struct initialization only happen when needed. :)